rasa_sdk/executor.py

```diff
@@ -33,19 +33,27 @@ def utter_message(
         image: Optional[Text] = None,
         json_message: Optional[Dict[Text, Any]] = None,
         template: Optional[Text] = None,
+        response: Optional[Text] = None,
         attachment: Optional[Text] = None,
         buttons: Optional[List[Dict[Text, Any]]] = None,
         elements: Optional[List[Dict[Text, Any]]] = None,
         **kwargs: Any,
     ) -> None:
-        """"Send a text to the output channel"""
-
+        """Send a text to the output channel."""
+        if template and not response:
```

Comment:
why not this?
```suggestion
        if template:
```

Reply:
I was thinking `response` should be the source of truth. If `template` was used, it still shouldn't overwrite `response` if it wasn't not empty. 🤷🏻‍♀️

Reply:
sounds good